codegen/CodeGen/Struct.pm

```diff
@@ -52,19 +52,33 @@ sub preprocess_struct_member {
 
     if ($mem->{type} eq "pad" and not exists $mem->{size}) {confess "pad type need a size"};
 
-    $mem->{array_size} //= 1;
+    $mem->{array_size} = exists $mem->{array_size} ? (ensure_number $mem->{array_size}) : (1);
```

Comment:
You can do that same trick we did before: `$mem->{array_size} = ensure_number( $mem->{array_size} // 1 );`

Reply:
Idk how I forgot about that